packages/office-addin-mock/src/officeMockObject.ts

```diff
@@ -0,0 +1,179 @@
+import { usageDataObject } from "./defaults";
+
+/**
+ * Creates an office-js mockable object
+ * @param object Object structure to provide initial values for the mock object (Optional)
+ */
+export class OfficeMockObject {
+  constructor(object?: ObjectData) {
+    this.properties = new Map<string, OfficeMockObject>();
+    this.loaded = false;
+    this.resetValue(undefined);
+    if (object) {
+      this.populate(object);
+    }
+  }
+
+  /**
+   * Adds a function to OfficeMockObject. The function can be accessed by simply calling `this.methodName`
+   * @param methodName Method name of the function to be added
+   * @param methodBody Function to be added to the object. A blank function will be added if no argument is provided (Optional)
+   */
+  addMockFunction(methodName: string, methodBody?: Function) {
+    this[methodName] = methodBody ? methodBody : function () {};
+  }
+
+  /**
+   * Adds a mock object to OfficeMockObject. The object can be accessed by simply calling `this.objectName`
+   * @param objectName Object name of the object to be added
+   */
+  addMock(objectName: string) {
+    const officeMockObject = new OfficeMockObject();
+    officeMockObject.isTreatedLikeObject = true;
+    this.properties.set(objectName, officeMockObject);
+    this[objectName] = this.properties.get(objectName);
+  }
+
+  /**
+   * Mock replacement of the load method in the Office.js API
+   * @param propertyArgument Argument of the load call. Will load any properties in the argument
+   */
+  load(propertyArgument: string | string[]) {
+    if (typeof propertyArgument === "string") {
+      this.loadMultipleProperties(propertyArgument);
+    } else {
+      propertyArgument.forEach((property: string) => {
+        this.loadMultipleProperties(property);
+      });
+    }
+  }
+
+  /**
+   * Adds a property of any type to OfficeMockObject
+   * @param propertyName Property name to the property to be added
+   * @param value Value this added property will have
+   */
+  setMock(propertyName: string, value: unknown) {
+    if (!this.properties.has(propertyName)) {
+      const officeMockObject = new OfficeMockObject();
+      officeMockObject.isTreatedLikeObject = false;
+      this.properties.set(propertyName, officeMockObject);
+    }
+    this.properties.get(propertyName)?.resetValue(value);
+    this[propertyName] = this.properties.get(propertyName)?.value;
+  }
+
+  /**
+   * Mock replacement for the sync of Office.js API
+   */
+  sync() {
+    this.properties.forEach((property: OfficeMockObject, key: string) => {
+      property.sync();
+      this.assignValue(key);
+    });
+    if (this.loaded) {
+      this.value = this.valueBeforeLoaded;
+    }
+  }
+
+  private assignValue(propertyName: string) {
```

Comment:
what's the value getting assigned?  I think the method name is confusing.

Reply:
I do agree. This method is to "assign" the property to be called using this.propertyName.
Changed the name to `makePropertyCallable`